src/main/java/org/example/LocalClasesAndAnonymousSorting/AnonymSorting.java

```diff
@@ -0,0 +1,52 @@
+package org.example.LocalClasesAndAnonymousSorting;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+
+public class AnonymSorting {
+
+    public static void main(String[] args) {
+
+        Employee e1 = new Employee("Minnie", "Mouse", "01/02/2015");
+        Employee e2 = new Employee("Mickey", "Mouse", "05/08/2000");
+        Employee e3 = new Employee("Daffy", "Duck", "11/02/2011");
+        Employee e4 = new Employee("Daisy", "Duck", "05/03/2013");
+        Employee e5 = new Employee("Goofy", "Dog", "23/07/2020");
+
+        List<Employee> list = new ArrayList<>(Arrays.asList(e1, e2, e3, e4, e5));
+
+        printOrderedList(list, "name");
+        System.out.println();
+
+        printOrderedList(list, "year");
+    }
+    public static void printOrderedList(List<Employee> list, String criteria) {
+        Collections.sort(list, new Comparator<Employee>() {
+            @Override
+            public int compare(Employee e1, Employee e2) {
+                if (criteria.equals("name")) {
+                    return e1.getFullName().compareTo(e2.getFullName());
+                } else if (criteria.equals("year")) {
+                    return e1.getYearsWorked() - e2.getYearsWorked();
+                }
+                return 0;
+            }
+        });
+
+        for (Employee employee : list) {
+            System.out.println(employee.getFullName() + " - Years Worked: " + employee.getYearsWorked());
```

Comment:
in part of printed the idea of the task was to use new local class which will accept Employee in constructor and transform it to print info in required format.
"Create a local class within the main method to encapsulate additional functionality for each employee, such as calculating years worked and constructing a full name.
Using the local class, create a list of employees with calculated fields."